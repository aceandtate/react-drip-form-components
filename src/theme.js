export const colors = {
  white: 'rgb(255, 255, 255)',
  grayDark: 'rgb(54, 54, 54)',
  gray: 'rgb(204, 204, 204)',
  grayLight: 'rgb(187, 187, 187)',
  grayLightest: 'rgb(219, 219, 219)',
  primary: 'rgb(0, 188, 212)',
  error: 'rgb(245, 0, 87)',
  disabled: 'rgb(245, 245, 245)',
  shadow: 'rgba(0, 0, 0, 0.04)',
};

export const theme = {
  // typography
  fontSize: '1rem',
  color: colors.grayDark,
  error: colors.error,
  disabled: colors.gray,

  // background
  background: colors.white,
  backgroundEnter: colors.white,
  backgroundFocus: colors.white,
  backgroundDisabled: colors.disabled,
  backgroundError: colors.white,

  // border
  borderWidth: '1px',
  borderRadius: '3px',
  borderColor: colors.grayLightest,
  borderColorEnter: colors.gray,
  borderColorFocus: colors.primary,
  borderColorDisabled: colors.disabled,
  borderColorError: colors.error,

  // box-shadow
  boxShadow: '0 0 0 transparent',
  boxShadowEnter: `0 2px 2px ${colors.shadow} inset`,
  boxShadowDisabled: `0 2px 2px ${colors.shadow} inset`,

  // transition
  transitionProperty: 'color, box-shadow, border',
  transitionDuration: '110ms',
  transitionTimingFunction: 'cubic-bezier(.25, .46, .45, .94)', // ease-out-quad

  // control
  controlHeight: '2.7em',

  // select
  selectArrowColor: colors.primary,

  // spinner
  spinnerColor: colors.primary,
  spinnerDuration: '500ms',
};

export const extendTheme = (extendsTheme = {}) => ({
  ...theme,
  ...extendsTheme,
});

export const themeProps = {
  theme: {
    rdf: theme,
  },
};

export const get = (props, key) => props.theme.rdf[key];

export const bind = key => props => get(props, key);
