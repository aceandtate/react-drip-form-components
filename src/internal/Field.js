import styled from 'styled-components';
import { themeProps, bind } from '../theme';

const Field = styled.div`
  display: block;
  position: relative;
  margin: 0;
  font-size: ${bind('fontSize')};
`;

Field.defaultProps = themeProps;

export default Field;
