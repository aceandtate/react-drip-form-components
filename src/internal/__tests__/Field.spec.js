import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { theme } from '../../theme';
import Field from '../Field';
import { mountWithTheme } from '../../__tests__/utils';


describe('Internal#<Field />', () => {
  test('Should be render children', () => {
    const element = <div id="foo">foo bar baz</div>;
    const wrapper = shallow(
      <Field>
        {element}
      </Field>
    );

    expect(wrapper.contains(element)).toBe(true);
  });


  test('Should be styled component', () => {
    const wrapper = shallow(<Field />);
    expect(wrapper).toHaveStyleRule('margin', '0');
    expect(wrapper).toHaveStyleRule('font-size', theme.fontSize);
  });


  test('Should be changed styled by theme', () => {
    const wrapper = mountWithTheme(<Field />, { fontSize: '5em' });
    expect(wrapper).toHaveStyleRule('font-size', '5em');
  });
});
