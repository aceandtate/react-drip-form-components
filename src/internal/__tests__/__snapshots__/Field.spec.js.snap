// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internal#<Field /> Should be styled component 1`] = `
.c0 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 1rem;
}

<div
  className="c0"
/>
`;
